client/db.go

```diff
@@ -252,78 +250,68 @@ func (db *DB) GetProto(key interface{}, msg gogoproto.Message) error {
-// The returned Result will contain a single row and Result.Err will indicate
-// success or failure.
+// Put sets the value for a key, returning the set key/value or an error.
 //
 // key can be either a byte slice, a string, a fmt.Stringer or an
 // encoding.BinaryMarshaler. value can be any key type or a proto.Message.
-func (db *DB) Put(key, value interface{}) (Result, error) {
-	return runOne(db, db.B.Put(key, value))
+func (db *DB) Put(key, value interface{}) (KeyValue, error) {
+	return runOneRow(db, db.B.Put(key, value))
 }
 
 // CPut conditionally sets the value for a key if the existing value is equal
 // to expValue. To conditionally set a value only if there is no existing entry
 // pass nil for expValue.
 //
-// The returned Result will contain a single row and Result.Err will indicate
-// success or failure.
-//
 // key can be either a byte slice, a string, a fmt.Stringer or an
 // encoding.BinaryMarshaler. value can be any key type or a proto.Message.
-func (db *DB) CPut(key, value, expValue interface{}) (Result, error) {
-	return runOne(db, db.B.CPut(key, value, expValue))
+func (db *DB) CPut(key, value, expValue interface{}) (KeyValue, error) {
+	return runOneRow(db, db.B.CPut(key, value, expValue))
 }
 
 // Inc increments the integer value at key. If the key does not exist it will
 // be created with an initial value of 0 which will then be incremented. If the
 // key exists but was set using Put or CPut an error will be returned.
 //
-// The returned Result will contain a single row and Result.Err will indicate
-// success or failure.
-//
 // key can be either a byte slice, a string, a fmt.Stringer or an
 // encoding.BinaryMarshaler.
-func (db *DB) Inc(key interface{}, value int64) (Result, error) {
-	return runOne(db, db.B.Inc(key, value))
+func (db *DB) Inc(key interface{}, value int64) (KeyValue, error) {
+	return runOneRow(db, db.B.Inc(key, value))
 }
 
 // Scan retrieves the rows between begin (inclusive) and end (exclusive).
 //
-// The returned Result will contain up to maxRows rows and Result.Err will
-// indicate success or failure.
+// The returned []KeyValue will contain up to maxRows elements.
 //
 // key can be either a byte slice, a string, a fmt.Stringer or an
 // encoding.BinaryMarshaler.
-func (db *DB) Scan(begin, end interface{}, maxRows int64) (Result, error) {
-	return runOne(db, db.B.Scan(begin, end, maxRows))
+func (db *DB) Scan(begin, end interface{}, maxRows int64) ([]KeyValue, error) {
+	r, err := runOneResult(db, db.B.Scan(begin, end, maxRows))
+	return r.Rows, err
 }
 
 // Del deletes one or more keys.
 //
-// Each key will have a corresponding row in the returned Result.
+// Each key will have a corresponding entry in the returned []KeyValue.
 //
 // key can be either a byte slice, a string, a fmt.Stringer or an
 // encoding.BinaryMarshaler.
-func (db *DB) Del(keys ...interface{}) (Result, error) {
-	return runOne(db, db.B.Del(keys...))
+func (db *DB) Del(keys ...interface{}) ([]KeyValue, error) {
```

Comment:
Again, why return a result here if it doesn't contain anything that wasn't in the input keys slice?